build/mergeallmodules.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+
+#Assumes that ./x64 and ./x86 exist and contain the arch-specific copies of a DSC resource module
+
+set -e
+
+zip -L >/dev/null 2>&1
+if [ $? != 0 ]; then
+        echo "zip must be installed"
+        exit 1
+fi
+
+unzip -Z >/dev/null 2>&1
+if [ $? != 0 ]; then
+        echo "unzip must be installed"
+        exit 1
+fi
+
+TargetDir="./target"
+x64dir="./x64"
+x86dir="./x86"
+TestSigningDir=$1
+
+if [ "$TestSigningDir" = "" ]
+then
+	TestSigningDir="../test/config/testsigning"
+fi
+
+if [ -d "$TestSigningDir" ]
+then
+	SigningKeyFilePath="$TestSigningDir/signingkeys.gpg"
+	SigningKeyPassphrase="$TestSigningDir/passphrase"
+
+	if [ ! -f "$SigningKeyFilePath" ]
+	then
+		echo "GPG key file is not found"
+		exit 1
+	fi
+
+	if [ ! -f "$SigningKeyPassphrase" ]
+	then
+		echo "GPG key passphrase is not found"
+		exit 1
+	fi
+else
+	echo "Signing key directory doesn't exit. $TestSigningDir"
```

Comment:
Suggest changing to something like:

`echo "Signing key directory doesn't exist at location: $TestSingingDir"`

This is clearer.